src/rhsm/utils.py

```diff
@@ -276,3 +280,148 @@ def wrapper(*args, **kwargs):
             sys.stderr = stderr
             devnull.close()
     return wrapper
+
+
+class StatusMessage:
+    """Class for temorary reporting.
+
+    While you can call 'print()' and 'clean()' methods directly, the easier
+    way is to use context manager: 'with StatusMessage("Doing stuff..."):'.
+
+    This object will print the description to stdout when called. When the
+    context manager exists (either because the code finished or because an
+    error got raised inside), printed output will be cleared before the
+    program continues with its execution. This ensures that the message will
+    disappear after it's no longer valid.
+    """
+    def __init__(self, description: Optional[str]):
+        if description is None:
+            description = "Transmitting data"
+        self.raw_text = description
+
+        CURSIVE = "\033[3m"
+        RESET = "\033[0m"
+
+        self.text = f"{CURSIVE}{self.raw_text}...{RESET}"
+
+        self.quiet = False
+        config = rhsm.config.get_config_parser()
+        if config.get("rhsm", "quiet") == "1":
+            self.quiet = True
+        if not sys.stdout.isatty():
+            self.quiet = True
+
+    def print(self):
+        if self.quiet:
+            return
+        print(self.text, end="\r")
+
+    def clean(self):
+        if self.quiet:
+            return
+        print(" " * len(self.text), end="\r")
+
+    def __enter__(self):
+        self.print()
+
+    def __exit__(self, error_type, error_value, traceback):
+        self.clean()
+        if error_type:
+            raise
+
+
+class StatusSpinnerStyle:
+    LINE: str = "|/-\\"
+    BRAILLE: str = "⠋⠙⠸⠴⠦⠇"
+    WIDE_BRAILLE: List[str] = ["⠧ ", "⠏ ", "⠋⠁", "⠉⠉", "⠈⠙", " ⠹", " ⠼", "⠠⠴", "⠤⠤", "⠦⠄"]
+    BAR_FORWARD: List[str] = ["[    ]", "[=   ]", "[==  ]", "[=== ]", "[====]", "[ ===]", "[  ==]", "[   =]"]
+    BAR_BACKWARD: List[str] = ["[    ]", "[   =]", "[  ==]", "[ ===]", "[====]", "[=== ]", "[==  ]", "[=   ]"]
+    BAR_BOUNCE: List[str] = BAR_FORWARD + BAR_BACKWARD
+
+    options = [
+        "LINE", "BRAILLE", "WIDE_BRAILLE",
+        "BAR_FORWARD", "BAR_BACKWARD", "BAR_BOUNCE",
+    ]
+
+
+class StatusSpinnerPlacement:
+    BEFORE: str = "BEFORE"
+    AFTER: str = "AFTER"
+
+    options = ["BEFORE", "AFTER"]
+
+
+class LiveStatusMessage(StatusMessage):
+    """Class for temporary reporting, with spinner to show activity."""
+
+    def __init__(
+        self,
+        description: Optional[str],
+        *,
+        style: StatusSpinnerStyle = StatusSpinnerStyle.BRAILLE,
+        placement: StatusSpinnerPlacement = StatusSpinnerPlacement.AFTER,
```

Comment:
`StatusSpinnerPlacement.BEFORE` IMHO looks better, because the animation is still on one position.